src/core/plugins/postmarketos/plugin.js

```diff
@@ -0,0 +1,125 @@
+"use strict";
+
+/*
+ * Copyright (C) 2022 Alexander Martinz <alexander@ubports.com>
+ * Copyright (C) 2022 Caleb Connolly <caleb@connolly.tech>
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+const Plugin = require("../plugin.js");
+const api = require("./api.js");
+const fs = require("fs-extra");
+const path = require("path");
+
+/**
+ * postmarketOS plugin
+ * @extends Plugin
+ */
+class PostmarketOsPlugin extends Plugin {
+  /**
+   * action download
+   * @returns {Promise<Array<Object>>}
+   */
+  action__download() {
+    return api
+      .getImages(
+        this.props.settings["release"],
+        this.props.settings["interface"],
+        this.props.config.codename
+      )
+      .then(files => [
+        {
+          actions: [
+            {
+              "core:download": {
+                group: "postmarketOS",
+                files
+              }
+            },
+            {
+              "core:unpack": {
+                group: "postmarketOS",
+                files: files.map(file => ({
+                  ...file,
+                  archive: file.name || path.basename(file.url),
+                  dir: "."
+                }))
+              }
+            },
+            {
+              "postmarketos:rename_unpacked_files": {
+                group: "postmarketOS",
+                files
+              }
+            }
+          ]
+        }
+      ]);
+  }
+
+  /**
+   * action rename_unpacked_files
+   * @returns {Promise<void>}
+   */
+  action__rename_unpacked_files({ group, files }) {
+    return Promise.resolve().then(() => {
```

Comment:
This function could be declared `async`, then you won't need the `Promise.resolve().then()` chain since those always return promises.